solutions/web3-authentication/pages/index.tsx

```diff
@@ -0,0 +1,240 @@
+import Head from 'next/head'
+import {
+  Layout,
+  Text,
+  Page,
+  Link,
+  Code,
+  Snippet,
+  Button,
+} from '@vercel/examples-ui'
+import { getCsrfToken, signIn, useSession, signOut } from 'next-auth/react'
+import { SiweMessage } from 'siwe'
+import { useAccount, useConnect, useNetwork, useSignMessage } from 'wagmi'
+
+function Home() {
+  const session = useSession()
+
+  const [{ data: connectData }, connect] = useConnect()
+  const [, signMessage] = useSignMessage()
+  const [{ data: networkData }] = useNetwork()
+  const [{ data: accountData }] = useAccount()
+
+  const handleLogin = async () => {
+    try {
+      await connect(connectData.connectors[0])
+      const callbackUrl = '/protected'
+      const message = new SiweMessage({
+        domain: window.location.host,
+        address: accountData?.address,
+        statement: 'Sign in with Ethereum to the app.',
+        uri: window.location.origin,
```

Comment:
Can we get this from the router? Just to not rely directly on window

Reply:
I checked the docs and it does not seem like it. 

We could have it come from a Env variable instead. What do you think?